src/hpc/thread.c

```diff
@@ -1285,4 +1323,78 @@ void ReadGuardError(Obj o)
 }
 #endif
 
+// These are temporary debugging functions.
+
+static Int NumReadErrors = 0;
+static Int NumWriteErrors = 0;
+volatile int GuardDummy;
+
+#ifdef DEBUG_GUARDS
+
+#include <execinfo.h>
+
+#define BACKTRACE_DEPTH 128
+
+void SetGuardErrorStack(void)
+{
+    void *  callstack[BACKTRACE_DEPTH];
+    int     depth = backtrace(callstack, BACKTRACE_DEPTH);
+    char ** calls = backtrace_symbols(callstack, depth);
+    Obj     stack = NEW_PLIST_IMM(T_PLIST, depth);
+    SET_LEN_PLIST(stack, depth - 1);
+    for (int i = 1; i < depth; i++) {
+        char * p = calls[i] + 1;
+        char * q = p;
+        while (*p) {
+            if (p[-1] == ' ' && p[0] == ' ')
+                p++;
+            else
+                *q++ = *p++;
+        }
+        *q++ = 0;
+        SET_ELM_PLIST(stack, i, MakeImmString(calls[i]));
```

Comment:
This is missing a `CHANGED_BAG` call, isn't it? Of course for Boehm GC that doesn't matter, but in the future we might be using the Julia GC here, no? Then it would matter.

Reply:
Fixed.